### ISSUE TITLE
entire codes

### ISSUE THREAD

**Issue by beihaiyou** [lkpengcs/DCDA#1]
Dear Linkai Peng
    I'm so interested in your work DCDA. But the the codes lack the the key part 'segmentation_models_pytorch',when I download your code in github. I would be so grateful if you give me your entire codes. 
    Looking forward to hearing from you

**Comment by lkpengcs** [comment]
Thank you for your interest! You can install the library via this [link](https://smp.readthedocs.io/en/latest/).

**Comment by beihaiyou** [comment]
Thank you so much for your reply, I will try it right now.

**Comment by wh-isper** [comment]
> Thank you so much for your reply, I will try it right now.

请问您代码跑通了吗？

**Comment by wh-isper** [comment]
> Thank you so much for your reply, I will try it right now.

hello，did you reimplement the results?

**Comment by beihaiyou** [comment]
NO, I use my own datas, I preprocess the datas following the paper, but the network dosen't converge. I don't know in which step I make mistakes, could you please give me some instruction? @lkpengcs 
following is my loss graph
![1677481951066](https://user-images.githubusercontent.com/93357283/221498473-92e03aaf-fdff-44ab-9db2-4de7a46cc833.png)
following is my data
![1677482484356](https://user-images.githubusercontent.com/93357283/221499224-4e70a24f-7887-41f3-8646-af99341aabc9.jpg)
![1677482506342](https://user-images.githubusercontent.com/93357283/221499300-a490fb0f-b77e-4427-b8ae-4c59bfbaff4a.jpg)

**Comment by lkpengcs** [comment]
It may not work on such datasets.

**Comment by wh-isper** [comment]
I used the polyp dataset, Kvasir, ETIS, and CVC-ColonDB. And I experienced several times, but none of them converge.

**Comment by beihaiyou** [comment]
I use CT and PET datas, it may be same as yours

**Comment by Djoels** [comment]
hello, I tried to reproduce results on the OCTA-500 dataset but without success yet.
can you provide the samples that were mentioned here?:

"We manually select and form a subset of OCTA-500 (191 from 3M and 254 from 6M), eliminating samples with
severe quality issues on either the OCT domain or the OCTA domain. To be noted, the OCT and OCTA images in our subset are unpaired. We resize each image to 384 × 384. In this subset, 50 images are used for testing and the others are used
for training."

**Comment by lkpengcs** [comment]
You can access the dataset via this [link](https://drive.google.com/drive/folders/13P7JgYh7J5nMZSckvdkkbiSlPkso-aBV?usp=share_link).

**Comment by Djoels** [comment]
thank you very much! will give this a go

**Comment by Doris116** [comment]
Dear Linkai Peng 
 I am very interested in the work you have done, but the reproduction failed, and there was an error of missing parameters. Another point is how is your dataset directory arranged?

**Comment by lkpengcs** [comment]
You need to manually set some required parameters.
The arrangement of the dataset is the same as the one in the above link.

**Comment by Doris116** [comment]
After configuring the parameters, an error is reported, indicating that the file is not found. How should I configure the parameters?

**Comment by lkpengcs** [comment]
Could you check if the path is correct?

**Comment by Doris116** [comment]
各位前辈，请问数据集路径是什么样的呢

**Comment by Doris116** [comment]
![image](https://github.com/lkpengcs/DCDA/assets/133087482/e5785d15-9dcf-413a-86af-01add61ec09f)
geq各位前辈，请问复现过程中报这种错误是为什么呢，应该如何修改呀

**Comment by lkpengcs** [comment]
Input channel of the image should be 1

**Comment by Doris116** [comment]
![image](https://github.com/lkpengcs/DCDA/assets/133087482/a5ed712b-773f-4704-977a-c94db70af927)
Thank you for your reply, however, it is strange why this error is reported again after changing the number of input channels to 1. There is a feeling of contradicting each other.

**Comment by Doris116** [comment]
![image](https://github.com/lkpengcs/DCDA/assets/133087482/4bd3410b-ee0e-4118-8780-745e8b22d5bf)
Thank you for your reply, however, it is strange why this error is reported again after changing the number of input channels to 1. There is a feeling of contradicting each other.

**Comment by lkpengcs** [comment]
In options.py, you need to set input_dim_a and input_dim_b to 1, not simply reshape the tensor.

**Comment by Doris116** [comment]
![Uploading c948f69f277422edaea6b9e16204949.png…]()
Thanks for the pointers, I encountered another error later on, is this error due to the wrong version? I changed the model.py from "a+=b" to "a= a+b" and it still doesn't work, is there a parameter that needs to be changed?

**Comment by Doris116** [comment]
![Uploading c948f69f277422edaea6b9e16204949.png…]()

**Comment by lkpengcs** [comment]
The picture you attached cannot be seen.
